source/client-side-encryption/tests/unified/rewrapManyDataKey-provider-kmip.yml

```diff
@@ -0,0 +1,167 @@
+description: rewrapManyDataKey-provider-kmip
+
+schemaVersion: "1.8"
+
+runOnRequirements:
+  - csfle: true
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents:
+        - commandStartedEvent
+  - clientEncryption:
+      id: &clientEncryption0 clientEncryption0
+      clientEncryptionOpts:
+        keyVaultClient: *client0
+        keyVaultNamespace: keyvault.datakeys
+        kmsProviders:
+          aws: { accessKeyId: { $$placeholder: 1 }, secretAccessKey: { $$placeholder: 1 } }
+          azure: { tenantId: { $$placeholder: 1 }, clientId: { $$placeholder: 1 }, clientSecret: { $$placeholder: 1 } }
+          gcp: { email: { $$placeholder: 1 }, privateKey: { $$placeholder: 1 } }
+          kmip: { endpoint: { $$placeholder: 1 } }
+          local: { key: { $$placeholder: 1 } }
```

Comment:
The KMS credentials are required for decrypting the data keys that are being rewrapped. Given each data key (`modifiedCount: 5`) has a respective KMS provider, they require appropriate KMS credentials to be loaded from the environment.

> would it be preferable to just have a single test file for rewrapManyDataKey instead of duplicating these blocks

If we do not mind the test runner spending a disproportionate amount of time (over 20 seconds) running tests within a single test file, then I can revert the split and put them into the same test file. The reason for duplication is to test the NxN matrix of KMS provider rewrapping possibilities (e.g. AWS -> AWS, AWS -> Azure, etc.). If this is considered overkill, I can reduce the number of keys required to be rewrapped per test.

Reply:
After out-of-band conversation, concluded merging tests into a single file to avoid duplication and test file maintenance overhead is preferable to idiosyncracies of current C driver test runner progress reporting behavior.